### PR TITLE
arrowhead refactoring

### PR DESCRIPTION
**Pull request by yuvalroth**
code cleanup

### CODE REVIEW

File: arrowhead/arrowhead.rb [hunk 1]
```diff
@@ -15,20 +15,24 @@ class Arrowhead
...
   def self.classify(region, shape)
     if CLASSIFICATIONS.include? region
-      shapes = CLASSIFICATIONS[region]
-      if shapes.include? shape
-        arrowhead = shapes[shape]
-        "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
-      else
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
+      classify_shape_for_specfic_region region, shape
```
Comment (jmtai) [inline arrowhead/arrowhead.rb]: Extracting this check to a new method is a good way to reduce the overall size of the `classify` method as well as reduce the overall arrowheadedness. However, in Ruby the convention is to prefer guard clauses over if-else constructs.

For example, you can write the first line of the function like this:
```ruby
raise "Unknown region, please provide a valid region." if CLASSIFICATIONS.include? region
```

File: arrowhead/arrowhead.rb [hunk 2]
```diff
@@ -15,20 +15,24 @@ class Arrowhead
...
     else
       raise "Unknown region, please provide a valid region."
     end
   end
+
+  private_class_method :new
+  def self.classify_shape_for_specfic_region(region, shape)
```
Comment (jmtai) [inline arrowhead/arrowhead.rb]: Oops! Typo here: `specfic`  :)

File: arrowhead/arrowhead.rb [hunk 3]
```diff
@@ -15,20 +15,24 @@ class Arrowhead
...
+    shapes = CLASSIFICATIONS[region]
+    if shapes.include? shape
+      "You have a(n) '#{shapes[shape]}' arrowhead. Probably priceless."
+    else
+      raise "Unknown shape value."\
```
Comment (jmtai) [inline arrowhead/arrowhead.rb]: Add a space before the `\` character.

**Comment by jmtai** [comment]
Nice job! Try refactoring the code using guard clauses and resubmit when you're done.

**Comment by djkotowski** [comment]
Guard clauses help you break execution in a single line.  They typically follow these formats:

```ruby
raise YourException, "message" if condition
# or
return value unless condition
# or
break if condition
```

File: arrowhead/arrowhead.rb [hunk 4]
```diff
@@ -2,32 +2,34 @@ class Arrowhead
   # This seriously belongs in a database.
   CLASSIFICATIONS = {
     far_west: {
-      notched: "Archaic Side Notch",
-      stemmed: "Archaic Stemmed",
-      lanceolate: "Agate Basin",
-      bifurcated: "Cody",
+      notched: 'Archaic Side Notch',
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Sometimes rubocop complains about this, but we generally don't care if you use single or double-quotes, as long as they make sense for what you're trying to accomplish.

File: arrowhead/arrowhead.rb [hunk 5]
```diff
@@ -2,32 +2,34 @@ class Arrowhead
...
-      shapes = CLASSIFICATIONS[region]
-      if shapes.include? shape
-        arrowhead = shapes[shape]
-        "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
-      else
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
-    else
-      raise "Unknown region, please provide a valid region."
+    unless CLASSIFICATIONS.include? region
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Wrap arguments in parentheses.

File: arrowhead/arrowhead.rb [hunk 6]
```diff
@@ -2,32 +2,34 @@ class Arrowhead
...
+      raise 'Unknown region, please provide a valid region.'
     end
+    classify_shape_for_specific_region region, shape
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Ditto above.

**Comment by djkotowski** [comment]
Looks good.  Take a look at the comments for future exercises.  No need for a resubmission.

**Pull request closed without merging**